src/main/java/frc/robot/utils/ModeEstimator.java

```diff
@@ -0,0 +1,143 @@
+package frc.robot.utils;
+
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+import frc.robot.Constants;
+import frc.robot.Constants.Mode;
+import frc.robot.subsystems.drive.DriveIO;
+import frc.robot.subsystems.vision.VisionUtil.VisionMeasurement;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Handles pose estimation for both real-time and replay modes of a swerve drive robot. This class
+ * wraps WPILib's SwerveDrivePoseEstimator and provides additional functionality for handling vision
+ * measurements and replaying recorded data.
+ */
+public class ModeEstimator {
+  private final DriveIO io;
+  private final SwerveDriveKinematics kinematics;
+  private final SwerveDrivePoseEstimator poseEstimator;
+  private final SwerveModulePosition[] modulePositions;
+
+  /**
+   * Constructs a new ModeEstimator.
+   *
+   * @param io The DriveIO interface for accessing real hardware or simulation
+   * @param modulePositions Initial positions of all swerve modules
+   */
+  public ModeEstimator(DriveIO io, SwerveModulePosition[] modulePositions) {
+    this.io = io;
+    this.modulePositions = modulePositions;
+    this.kinematics = new SwerveDriveKinematics(Constants.SWERVE_MODULE_OFFSETS);
+    this.poseEstimator =
+        new SwerveDrivePoseEstimator(
+            kinematics, Rotation2d.fromDegrees(0), modulePositions, new Pose2d());
+  }
+
+  /**
+   * Resets the pose estimator to a specific pose. In replay mode, resets the internal pose
+   * estimator. In real-time mode, delegates to the IO interface.
+   *
+   * @param pose The pose to reset to
+   */
+  public void resetPose(Pose2d pose) {
+    if (Constants.currentMode == Mode.REPLAY) {
+      poseEstimator.resetPose(pose);
+    } else {
+      io.resetPose(pose);
+    }
+  }
+
+  /**
+   * Samples the pose at a specific timestamp.
+   *
+   * @param timestamp The timestamp to sample at (using WPILib timestamp)
+   * @return Optional containing the pose at the given timestamp, or empty if not available
+   */
+  public Optional<Pose2d> samplePoseAt(double timestamp) {
+    return Constants.currentMode == Mode.REPLAY
```

Comment:
why the style chnage from the above method? inline if here but not in resetPose? make the code consistent.

Reply:
You can inline on a return not on a logic check